truss/util/env_vars.py

```diff
@@ -0,0 +1,40 @@
+import os
+from typing import Dict, Optional
+
+
+class override_env_vars:
+    """A context manager for temporarily overwriting environment variables.
+
+    Usage:
+        with override_env_vars({'API_KEY': 'test_key', 'DEBUG': 'true'}):
+            # Environment variables are modified here
+            ...
+        # Original environment is restored here
+    """
+
+    def __init__(self, env_vars: Dict[str, str]):
+        """
+        Args:
+            env_vars: Dictionary of environment variables to set
+        """
+        self.env_vars = env_vars
+        self.original_vars: Dict[str, Optional[str]] = {}
+
+    def __enter__(self):
+        for key in self.env_vars:
+            self.original_vars[key] = os.environ.get(key)
```

Comment:
What about `dict(os.environ)`?

Reply:
it's not quite the same -- we only need to keep the values of env vars being overwritten, not the whole thing